### PR TITLE
Add github workflow to upload stable tarball

### PR DESCRIPTION
**Pull request by thefloweringash**
Context https://github.com/NixOS/nixpkgs/issues/104580#issuecomment-731724514

cc @LnL7

### CODE REVIEW

File: .github/workflows/upload-release-asset.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Adapted from the example workflow at
+# https://github.com/actions/upload-release-asset
+
+on:
+  push:
+    # Sequence of patterns matched against refs/tags
+    tags:
+    - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
```
Comment (LnL7) [inline .github/workflows/upload-release-asset.yml:8]: The idea here is to version this separately from the macOS versions right? So v0.1 gets an artifact for 10.12, 10.13, etc.

File: .github/workflows/upload-release-asset.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version: [ '10.12' ]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Create tarball
+        run: |
+          tar -cf stubs-${{ matrix.version }}.tar.gz -C stubs ${{ matrix.version }}
```
Comment (LnL7) [inline .github/workflows/upload-release-asset.yml]: This probably needs a few extra flsgs to make the output reproducible.

**Review by LnL7 — COMMENTED** [review]
Looks pretty good to me. I'm not sure yet if it might be better to fold this into bootstrap tools at some point, but for now something like this is a bit easier to update until things stabilize.

**Comment by thefloweringash** [comment]
Updated with options to make the tarball reproducible.

**Review by LnL7 — APPROVED** [review]

**Pull request merged**